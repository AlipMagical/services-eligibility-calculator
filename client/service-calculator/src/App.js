/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import './i18n';

import React, { createContext, useState } from 'react';

import { ANALYTICS_ID } from './common/config';
import { Body } from './components/Body';
import { DataContext } from './DataContext';
import Footer from './components/Footer';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import { Router } from 'react-router-dom';
import ScrollToTop from './common/utils/ScrollToTop';
import { createBrowserHistory } from 'history';

export const FilterContext = createContext();

export const createInitialFilterState = () => ({
  income: undefined,
  interests: {},
  people: {}
});

const history = createBrowserHistory();

export default function App(props) {
  const [analyticsInitialized, setAnalyticsInitialized] = useState(false);
  const [lastSearch, setLastSearch] = useState('');
  const [filterState, setFilterState] = useState(createInitialFilterState());
  const resetFilterState = () => setFilterState(createInitialFilterState());

  if (!analyticsInitialized) {
    ReactGA.initialize(ANALYTICS_ID, { testMode: props.testMode });
    setAnalyticsInitialized(true);
  }

  return (
    <DataContext.StateProvider>
      <FilterContext.Provider
        value={{
          filterState,
          setFilterState,
          resetFilterState,
          lastSearch,
          setLastSearch
        }}
      >
        <Router history={history}>
          <ScrollToTop />
          <Body />
          <Footer />
        </Router>
      </FilterContext.Provider>
    </DataContext.StateProvider>
  );
}

App.propTypes = {
  testMode: PropTypes.bool
};
