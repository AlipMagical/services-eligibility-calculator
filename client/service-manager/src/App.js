/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import './i18n';

import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import React, { createContext, useState } from 'react';
import {
  getLoggedInUser,
  getMillisUntilExpiration,
  isLoggedIn
} from './common/auth';

import { ANALYTICS_ID } from './common/config';
import { BASEPATH } from './common/config';
import Header from './components/Header';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import Toast from './components/Toast';
import departmentRoutes from './routes/department';
import getUserConfirmation from './getUserConfirmation';
import interestRoutes from './routes/interest';
import serviceRoutes from './routes/service';
import settingsRoutes from './routes/settings';
import { useSettings } from './hooks';

export const AuthContext = createContext();
export const ToastContext = createContext();
export const SettingsContext = createContext();

export default function App(props) {
  ReactGA.initialize(ANALYTICS_ID, { testMode: props.testMode });

  const [loginState, setLoginState] = useState({
    isLoggedIn: isLoggedIn(),
    loggedInUser: getLoggedInUser()
  });

  const onLoginStatusChanged = () => {
    //setTimeout to update when token might expire.
    const millis = getMillisUntilExpiration();
    millis && setTimeout(onLoginStatusChanged, millis);

    setLoginState({
      isLoggedIn: isLoggedIn(),
      loggedInUser: getLoggedInUser()
    });
  };

  const { settings, setSettings } = useSettings();

  const [toastMessage, setToastMessage] = useState(null);

  return (
    <AuthContext.Provider value={{ ...loginState, onLoginStatusChanged }}>
      <SettingsContext.Provider value={{ settings, setSettings }}>
        <ToastContext.Provider value={{ toastMessage, setToastMessage }}>
          <BrowserRouter getUserConfirmation={getUserConfirmation}>
            <Header />
            <div className="container bd-content">
              <Switch>
                {serviceRoutes}
                {settingsRoutes}
                {interestRoutes}
                {departmentRoutes}
                <Redirect from={BASEPATH} exact to={BASEPATH + 'services/'} />
                <Route render={() => <h1>Not Found</h1>} />
              </Switch>
            </div>
            {toastMessage && <Toast message={toastMessage} />}
          </BrowserRouter>
        </ToastContext.Provider>
      </SettingsContext.Provider>
    </AuthContext.Provider>
  );
}

App.propTypes = {
  testMode: PropTypes.bool
};
