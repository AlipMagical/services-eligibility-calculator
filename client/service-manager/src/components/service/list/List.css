/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.search-match {
  font-weight: bold;
}

.service-draft-name {
  font-size: 18px;
}

.draft-shared-by {
  margin-bottom: 0px;
}

.draft-shared-by > i {
  margin-right: 10px;
}

.draft-last-updated {
  color: #80868b;
  margin-bottom: 0px;
}

.service-draft-row {
  /* Rectangle 49 */

  width: 100%;
  height: 72px;

  /* grey/white (#fff) */

  background: #ffffff;
  /* grey/300 ✔ */

  border: 1px solid #dadce0;
  box-sizing: border-box;
  border-radius: 8px;

  border-left-color: #ffa500;
  border-left-width: 4px;

  margin-left: auto;
  margin-right: auto;

  padding-left: 9px;
  padding-right: 9px;
}

.service-draft-row.empty {
  border-left-color: #dadce0;
  border-left-width: 1px;
}

h1.list {
  margin-bottom: 40px;
}

h2.list {
  font-family: Google Sans;
  font-size: 18px;
  line-height: 24px;
  font-weight: normal;
  /* identical to box height, or 133% */

  /* grey/800 ✔ */

  color: #3c4043;
}

table.table td {
  height: 60px;
}

div.search.input-group {
  width: 250px;
}

.search.input-group span.input-group-text {
  background-color: #fff;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  padding-right: 0;
  border-right: none;
}

.search.input-group:focus-within {
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
  border-radius: 0.3em;
}

.search.input-group label {
  display: flex;
  margin: 0;
}

.search.input-group input {
  border-left: none;
  margin: 0;
}

.search.input-group input:focus {
  border-color: #ced4da;
  box-shadow: none;
}
