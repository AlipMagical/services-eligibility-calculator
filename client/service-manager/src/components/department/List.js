/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useContext, useState } from 'react';
import { defaultLocale, localeLanguage } from '../../Locales.js';

import { AuthContext } from '../../App.js';
import Card from '../../common/components/service/Card.js';
import Create from './Create';
import Show from './Show';
import Update from './Update';
import { base64URLEncode } from '../../common/base64URL';
import { hasRole } from '../../common/utils/index.js';
import { nonDefaultLocales } from '../namedTranslatedEntity.js';
import { useTranslatedDepartments } from '../../hooks';

function ListRow({ department, translations, onManage }) {
  return (
    <div
      key={department['@id']}
      className="d-flex justify-content-between mb-3"
    >
      <div>
        <span>{department['name']}</span>
        {translations && (
          <>
            <span className="ml-1">
              (
              {Object.entries(translations).map(([locale, translation]) => (
                <i key={locale}>
                  {localeLanguage(locale)}: {translation['name']}
                </i>
              ))}
              )
            </span>
          </>
        )}
      </div>{' '}
      <div>
        {onManage && (
          <button className="btn btn-blue-text pull-right" onClick={onManage}>
            Manage
          </button>
        )}
      </div>
    </div>
  );
}

export default function List() {
  const { departments, fetchDepartments } = useTranslatedDepartments();
  const { loggedInUser } = useContext(AuthContext);
  const isAdmin = hasRole(loggedInUser, 'ROLE_ADMIN');
  const [showCreate, setShowCreate] = useState(false);
  const [showEdit, setShowEdit] = useState(false);
  const [departmentId, setDepartmentId] = useState(undefined);

  return (
    <Card
      header={
        <div className="d-flex justify-content-between">
          <div>Departments</div>
          <div>
            {isAdmin && (
              <button
                className="btn btn-primary"
                onClick={() => setShowCreate(true)}
              >
                Add department
              </button>
            )}
          </div>
        </div>
      }
    >
      {showCreate && (
        <Create
          onCancel={() => {
            setShowCreate(false);
            fetchDepartments();
          }}
        />
      )}
      {departmentId && !showEdit && (
        <Show
          onCancel={() => {
            setDepartmentId(undefined);
            fetchDepartments();
          }}
          departmentId={departmentId}
          onEdit={() => setShowEdit(true)}
        />
      )}
      {departmentId && showEdit && (
        <Update
          departmentId={departmentId}
          onCancel={() => {
            setShowEdit(false);
            fetchDepartments();
          }}
          onRemove={() => {
            // Fully unwind the stack of modals so none of them try to
            // display the deleted item.
            setShowEdit(false);
            setDepartmentId(undefined);
          }}
        />
      )}
      {departments &&
        departments[defaultLocale] &&
        departments[defaultLocale].map(department => (
          <ListRow
            key={department['@id']}
            department={department}
            translations={nonDefaultLocales(department['@id'], departments)}
            onManage={
              isAdmin &&
              (() => {
                setDepartmentId(base64URLEncode(department['@id']));
              })
            }
          />
        ))}
    </Card>
  );
}
