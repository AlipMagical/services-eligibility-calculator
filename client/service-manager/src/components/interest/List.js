/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useContext, useState } from 'react';
import { defaultLocale, localeLanguage } from '../../Locales.js';

import { AuthContext } from '../../App.js';
import Card from '../../common/components/service/Card.js';
import Create from './Create';
import Show from './Show';
import Update from './Update';
import { base64URLEncode } from '../../common/base64URL';
import { hasRole } from '../../common/utils/index.js';
import { nonDefaultLocales } from '../namedTranslatedEntity.js';
import { useTranslatedInterests } from '../../hooks';

function ListRow({ interest, translations, onManage }) {
  return (
    <div key={interest['@id']} className="d-flex justify-content-between mb-3">
      <div>
        <i className="material-icons align-middle">
          {interest['materialIcon']}
        </i>
        <span className="align-middle ml-3">{interest['name']}</span>
        {translations && (
          <>
            <span className="align-middle ml-1">
              (
              {Object.entries(translations).map(([locale, translation]) => (
                <i key={locale}>
                  {localeLanguage(locale)}: {translation['name']}
                </i>
              ))}
              )
            </span>
          </>
        )}
      </div>
      <div>
        {onManage && (
          <button className="btn btn-blue-text" onClick={onManage}>
            Manage
          </button>
        )}
      </div>
    </div>
  );
}

export default function List() {
  const { interests, fetchInterests } = useTranslatedInterests();
  const { loggedInUser } = useContext(AuthContext);
  const isAdmin = hasRole(loggedInUser, 'ROLE_ADMIN');
  const [showCreate, setShowCreate] = useState(false);
  const [showEdit, setShowEdit] = useState(false);
  const [interestId, setInterestId] = useState(undefined);

  return (
    <Card
      header={
        <div className="d-flex justify-content-between mb-0">
          <div>Interests</div>
          <div>
            {isAdmin && (
              <button
                className="btn btn-primary"
                onClick={() => setShowCreate(true)}
              >
                Add interest
              </button>
            )}
          </div>
        </div>
      }
    >
      {showCreate && (
        <Create
          onCancel={() => {
            setShowCreate(false);
            fetchInterests();
          }}
        />
      )}
      {interestId && !showEdit && (
        <Show
          onCancel={() => {
            setInterestId(undefined);
            fetchInterests();
          }}
          interestId={interestId}
          onEdit={() => setShowEdit(true)}
        />
      )}
      {interestId && showEdit && (
        <Update
          interestId={interestId}
          onCancel={() => {
            setShowEdit(false);
            fetchInterests();
          }}
          onRemove={() => {
            // Fully unwind the stack of modals so none of them try to
            // display the deleted item.
            setShowEdit(false);
            setInterestId(undefined);
          }}
        />
      )}
      {interests &&
        interests[defaultLocale] &&
        interests[defaultLocale].map(interest => (
          <ListRow
            key={interest['@id']}
            interest={interest}
            translations={nonDefaultLocales(interest['@id'], interests)}
            onManage={
              isAdmin &&
              (() => {
                setInterestId(base64URLEncode(interest['@id']));
              })
            }
          />
        ))}
    </Card>
  );
}
