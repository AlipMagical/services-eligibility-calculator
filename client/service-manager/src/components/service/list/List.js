/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useContext, useState } from 'react';
import { useArchivedServices, useServices } from '../../../hooks';

import { AuthContext } from '../../../App';
import DraftsList from './DraftsList';
import { Helmet } from 'react-helmet';
import { Navbar } from 'react-bootstrap';
import SearchBar from './Search';
import ServicesTable from './ServicesTable';

export default function ListServices() {
  const { services, fetchServices } = useServices();
  const { archivedServices, fetchArchivedServices } = useArchivedServices();
  const { isLoggedIn } = useContext(AuthContext);
  const [wasloggedIn, setWasLoggedIn] = useState(isLoggedIn);
  const [lastSearch, setLastSearch] = useState('');

  if (isLoggedIn !== wasloggedIn) {
    setWasLoggedIn(isLoggedIn);
    fetchServices();
    fetchArchivedServices();
  }

  return (
    <>
      <Helmet>
        <title>{process.env.REACT_APP_MANAGER_TITLE}</title>
      </Helmet>
      <Navbar className="service-header">
        <div className="w-100">
          <h1 className="list float-left">
            {process.env.REACT_APP_MANAGER_TITLE}
          </h1>
          <SearchBar onSearch={setLastSearch} />
        </div>
      </Navbar>

      {isLoggedIn && <DraftsList lastSearch={lastSearch} />}
      {services && (
        <ServicesTable services={services} lastSearch={lastSearch} />
      )}
      {archivedServices && (
        <ServicesTable
          services={archivedServices}
          lastSearch={lastSearch}
          archived
        />
      )}
    </>
  );
}
