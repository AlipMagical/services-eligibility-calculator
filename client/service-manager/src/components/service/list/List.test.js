/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { AuthContext, SettingsContext } from '../../../App';
import { axe, toHaveNoViolations } from 'jest-axe';
import {
  mockArchivedServicesJson,
  mockServiceDraftsJson,
  mockServicesJson
} from '../../../../../testing/mockData';

import List from './List';
import { MemoryRouter } from 'react-router';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { getSettings } from '../../../../../testing/testHelpers';
import { render } from 'react-dom';

expect.extend(toHaveNoViolations);

it('renders list services view', async () => {
  const container = document.createElement('div');

  fetch.resetMocks();
  fetch
    .doMockOnceIf(
      url => url.includes('/api/service_drafts'),
      mockServiceDraftsJson
    )
    .doMockOnceIf(
      url =>
        url.includes(
          '/api/services?pagination=false&exists[archivedDateTime]=false'
        ),
      mockServicesJson
    )
    .doMockOnceIf(
      url =>
        url.includes(
          '/api/services?pagination=false&exists[archivedDateTime]=true'
        ),
      mockArchivedServicesJson
    );

  await act(async () => {
    render(
      <MemoryRouter>
        <SettingsContext.Provider value={getSettings()}>
          <AuthContext.Provider
            value={{
              isLoggedIn: true,
              loggedInUser: { username: 'user' }
            }}
          >
            <List match={{ params: {} }} />
          </AuthContext.Provider>
        </SettingsContext.Provider>{' '}
      </MemoryRouter>,
      container
    );
  });

  expect(container).toMatchSnapshot();
  expect(await axe(container)).toHaveNoViolations();
  expect(container.innerHTML).toContain('You');
});
