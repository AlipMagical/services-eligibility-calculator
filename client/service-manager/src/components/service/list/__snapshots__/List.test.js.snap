// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list services view 1`] = `
<div>
  <nav
    class="service-header navbar navbar-expand navbar-light"
  >
    <div
      class="w-100"
    >
      <h1
        class="list float-left"
      >
        Manage city services
      </h1>
      <div
        class="input-group search float-right"
      >
        <div
          class="input-group-prepend"
        >
          <label
            for="search-input"
          >
            <span
              aria-hidden="true"
              class="input-group-text"
            >
              <i
                class="material-icons"
              >
                search
              </i>
            </span>
          </label>
        </div>
        <input
          aria-label="Search"
          class="form-control"
          id="search-input"
          placeholder="Search"
          type="text"
        />
      </div>
    </div>
  </nav>
  <h2
    class="list float-left"
  >
    Your
     draft services (
    2
    )
  </h2>
  <div
    class="float-right"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <a
          class="btn btn-primary pull-right"
          href="/servicedrafts/create"
          role="button"
        >
          Add service
        </a>
      </div>
    </div>
  </div>
  <div
    class="mb-5"
  >
    <div
      class="row service-draft-row my-2"
    >
      <div
        class="col-6 my-auto service-draft-name"
      >
        <span>
          A New Draft
        </span>
      </div>
      
      <div
        class="col my-auto"
      >
        <a
          class="btn btn-blue-text pull-right"
          href="/servicedrafts/show/L2FwaS9zZXJ2aWNlX2RyYWZ0cy8x"
          role="button"
        >
          Manage
        </a>
      </div>
    </div>
    <div
      class="row service-draft-row my-2"
    >
      <div
        class="col-6 my-auto service-draft-name"
      >
        <span>
          Draft for New Service
        </span>
      </div>
      <div
        class="col my-auto"
      >
        <div
          class="row draft-last-updated"
        >
          Last updated 
          20 days ago
           byÂ 
          <span>
            You
          </span>
        </div>
      </div>
      <div
        class="col my-auto"
      >
        <a
          class="btn btn-blue-text pull-right"
          href="/servicedrafts/show/L2FwaS9zZXJ2aWNlX2RyYWZ0cy8y"
          role="button"
        >
          Manage
        </a>
      </div>
    </div>
  </div>
  <h2
    class="list"
  >
    All published services
     (
    19
    )
  </h2>
  <table
    class="table table-fixed"
  >
    <thead>
      <tr>
        <th
          style="cursor: pointer; width: 30%;"
        >
          name
          <i
            class="material-icons"
          >
            arrow_downward
          </i>
        </th>
        <th
          style="cursor: pointer; width: 20%;"
        >
          department
        </th>
        <th
          style="cursor: pointer; width: 20%;"
        >
          last modified by
        </th>
        <th
          style="cursor: pointer; width: 15%;"
        >
          last modified
        </th>
        <th
          style="width: 15%;"
        />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <span>
            Accusantium aperiam consequatur.
          </span>
        </td>
        <td>
          <span>
            Housing and Human Services
          </span>
        </td>
        <td>
          <span>
            Mrs. Raphaelle Schumm DVM
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8w"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            At consequatur doloribus voluptatem voluptatem.
          </span>
        </td>
        <td>
          <span>
            Fire
          </span>
        </td>
        <td>
          <span>
            Alexandrea Mayer
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8x"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            At corporis facilis inventore vero animi.
          </span>
        </td>
        <td>
          <span>
            Police
          </span>
        </td>
        <td>
          <span>
            Domenico Kris
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8y"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Aut quibusdam labore sit.
          </span>
        </td>
        <td>
          <span>
            Planning
          </span>
        </td>
        <td>
          <span>
            Bernie Lockman
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8z"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Corrupti cum ratione animi.
          </span>
        </td>
        <td>
          <span>
            OSMP
          </span>
        </td>
        <td>
          <span>
            Beaulah Schuster
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy80"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Dolores ut natus et.
          </span>
        </td>
        <td>
          <span>
            Innovation and Technology
          </span>
        </td>
        <td>
          <span>
            Sigurd Bergnaum
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy81"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Et omnis eaque pariatur non.
          </span>
        </td>
        <td>
          <span>
            City Attorney
          </span>
        </td>
        <td>
          <span>
            Barry Schneider Sr.
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy82"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Et quam ut voluptatem qui.
          </span>
        </td>
        <td>
          <span>
            Planning
          </span>
        </td>
        <td>
          <span>
            Keyshawn Emmerich Jr.
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy83"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Eum porro est doloremque.
          </span>
        </td>
        <td>
          <span>
            City Clerk's Office
          </span>
        </td>
        <td>
          <span>
            Alyson Klocko
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy84"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Food Tax Rebate Program
          </span>
        </td>
        <td>
          <span>
            Housing and Human Services
          </span>
        </td>
        <td>
          <span>
            You
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy85"
            role="button"
          >
            Manage
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Modi cumque amet.
          </span>
        </td>
        <td>
          <span>
            City Manager's Office
          </span>
        </td>
        <td>
          <span>
            Dr. Christopher McGlynn DVM
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8xMA"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Officiis excepturi eligendi ducimus autem quo.
          </span>
        </td>
        <td>
          <span>
            City Council
          </span>
        </td>
        <td>
          <span>
            Cruz Carter DDS
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8xMQ"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Perferendis sit veniam.
          </span>
        </td>
        <td>
          <span>
            Public Works
          </span>
        </td>
        <td>
          <span>
            Ignatius Schimmel
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8xMg"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Provident non incidunt magnam molestias.
          </span>
        </td>
        <td>
          <span>
            Housing and Human Services
          </span>
        </td>
        <td>
          <span>
            Dandre Hilpert Jr.
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8xMw"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Quidem optio occaecati.
          </span>
        </td>
        <td>
          <span>
            Municipal Court
          </span>
        </td>
        <td>
          <span>
            Charity Jacobson
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8xNA"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Recusandae asperiores accusamus nihil repellat.
          </span>
        </td>
        <td>
          <span>
            Innovation and Technology
          </span>
        </td>
        <td>
          <span>
            Nicholas Berge
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8xNQ"
            role="button"
          >
            Manage
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Senior Center Fitness Discounts
          </span>
        </td>
        <td>
          <span>
            Housing and Human Services
          </span>
        </td>
        <td>
          <span>
            Jane Smith
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8xNg"
            role="button"
          >
            Manage
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Tempora aut eum et blanditiis.
          </span>
        </td>
        <td>
          <span>
            OSMP
          </span>
        </td>
        <td>
          <span>
            Dr. Reese Bahringer Jr.
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8xNw"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Voluptas iusto animi consequatur.
          </span>
        </td>
        <td>
          <span>
            OSMP
          </span>
        </td>
        <td>
          <span>
            Adaline Wyman IV
          </span>
        </td>
        <td>
          20 days ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8xOA"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <h2
    class="list"
  >
    Archived services
     (
    1
    )
  </h2>
  <table
    class="table table-fixed"
  >
    <thead>
      <tr>
        <th
          style="cursor: pointer; width: 30%;"
        >
          name
          <i
            class="material-icons"
          >
            arrow_downward
          </i>
        </th>
        <th
          style="cursor: pointer; width: 20%;"
        >
          department
        </th>
        <th
          style="cursor: pointer; width: 20%;"
        >
          archived by
        </th>
        <th
          style="cursor: pointer; width: 15%;"
        >
          archived
        </th>
        <th
          style="width: 15%;"
        />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <span>
            Aut soluta in.
          </span>
        </td>
        <td>
          <span>
            City Clerk's Office
          </span>
        </td>
        <td>
          <span>
            Jane Doe
          </span>
        </td>
        <td>
          7 years ago
        </td>
        <td>
          <a
            class="btn btn-blue-text pull-right"
            href="/show/L2FwaS9zZXJ2aWNlcy8xOQ"
            role="button"
          >
            View
          </a>
        </td>
      </tr>
    </tbody>
  </table>
   
</div>
`;
